pkg/api/handler/default.go

```diff
@@ -332,12 +334,26 @@ func getTransactionOptions(params api.POSTTransactionParams) (*metamorph.Transac
 	return getTransactionsOptions(api.POSTTransactionsParams(params))
 }
 
+func ValidateCallbackURL(callbackURL string) error {
+	_, err := url.ParseRequestURI(callbackURL)
+	if err != nil {
+		return fmt.Errorf("invalid callback URL [%w]", err)
+	}
+
+	rejectedCallbackUrlSubstrings := viper.GetStringSlice("metamorph.rejectCallbackContaining")
```

Comment:
We shouldn't use viper here. Like in the other services this setting should be passed as an option https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/pkg/api/handler/default.go#L42

when a new default handler is created https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/cmd/arc/services/api.go#L109

This should be called `viper.GetStringSlice("metamorph.rejectCallbackContaining")` in `api.go`

Reply:
For now no option is passed here `handler.NewDefault(logger, metamorphClient, policy)` How would it work? Would I pass function option which, after running, would set local slice of unacceptable callback url strings(read from viper)?

Reply:
Yes. Options are already defined here: https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/pkg/api/handler/default.go#L42

Slice of unacceptable callback url strings should then be a member of the `*ArcDefaultHandler` - type and passed in as option read from Viper